File: ResQNet_Project/resqnet-backend/src/main/java/com/pu_deltaforce/resqnet_backend/model/MissingPerson.java
package com.pu_deltaforce.resqnet_backend.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Entity
@Table(name = "missing_persons")
@Data
public class MissingPerson {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private Integer age;

    @Enumerated(EnumType.STRING)
    private Gender gender;

    @Column(nullable = false)
    private String lastSeenLocation;

    @Column(length = 1000)
    private String description;

    @Column(nullable = false)
    private String reporterName;

    @Column(nullable = false)
    private String reporterPhone;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(nullable = false)
    private LocalDateTime reportedAt;

    public enum Gender {
        MALE, FEMALE, OTHER
    }

    public enum Status {
        MISSING, FOUND
    }

    @PrePersist
    public void prePersist() {
        this.status = Status.MISSING;
        this.reportedAt = LocalDateTime.now();
    }
}

File: ResQNet_Project/resqnet-frontend/src/app/core/services/missing-person.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { environment } from '../../../environments/environment';

export interface MissingPerson {
  id?: number;
  name: string;
  age: number;
  gender: 'MALE' | 'FEMALE' | 'OTHER';
  lastSeenLocation: string;
  description: string;

  reporterName: string;
  reporterPhone: string;

  status?: 'MISSING' | 'FOUND';
  reportedAt?: string;
}

@Injectable({ providedIn: 'root' })
export class MissingPersonService {
  private apiUrl = `${environment.apiUrl}/missing-persons`;

  constructor(private http: HttpClient) { }

  getAll() { return this.http.get<MissingPerson[]>(this.apiUrl); }
  create(data: MissingPerson) { return this.http.post<MissingPerson>(this.apiUrl, data); }
  markFound(id: number) { return this.http.put(`${this.apiUrl}/${id}/found`, {}); }
  searchByName(name: string) { return this.http.get<MissingPerson[]>(`${this.apiUrl}/search/name/${name}`); }
  searchByLocation(loc: string) { return this.http.get<MissingPerson[]>(`${this.apiUrl}/search/location/${loc}`); }
}